### PR TITLE
Add `assertion-message` rule (fixes #59)

### LINKED ISSUE #59: assertion-message rule
Require or disallow the message parameter being used for assertions.
With options for `always` require or to `never` allow.
Default to `always`.

# Option Always
## Good
``` js
test(t => {
    t.pass('is passing');
    t.fail('is failing');
    t.ok('unicorn', 'is unicorn');
    t.notOk('unicorn', 'is not unicorn');
    t.true(true, 'is true');
    t.false(false, 'is false');
    t.is('same', 'same', 'is same');
    t.not('not', 'same', 'is not same');
    t.deepEqual({}, {}, 'is same object'); 
    t.notDeepEqual({}, {a: true}, 'is name same object'); 
    t.throws(Promise.reject(), 'is exception');
    t.notThrows(Promise.resolve(), 'is not exception');
    t.regex(a, /a/, 'is matching pattern');
    t.ifError(new Error(), 'is error');
});
```

## Bad
``` js
test(t => {
    t.pass();
    t.fail();
    t.ok('unicorn');
    t.notOk('unicorn');
    t.true(true);
    t.false(false);
    t.is('same', 'same');
    t.not('not', 'same');
    t.deepEqual({}, {}); 
    t.notDeepEqual({}, {a: true}); 
    t.throws(Promise.reject());
    t.notThrows(Promise.resolve());
    t.regex(a, /a/);
    t.ifError(new Error());
});
```

# Option Never

## Good
``` js
test(t => {
    t.pass();
    t.fail();
    t.ok('unicorn');
    t.notOk('unicorn');
    t.true(true);
    t.false(false);
    t.is('same', 'same');
    t.not('not', 'same');
    t.deepEqual({}, {}); 
    t.notDeepEqual({}, {a: true}); 
    t.throws(Promise.reject());
    t.notThrows(Promise.resolve());
    t.regex(a, /a/);
    t.ifError(new Error());
});
```

## Bad
``` js
test(t => {
    t.pass('is passing');
    t.fail('is failing');
    t.ok('unicorn', 'is unicorn');
    t.notOk('unicorn', 'is not unicorn');
    t.true(true, 'is true');
    t.false(false, 'is false');
    t.is('same', 'same', 'is same');
    t.not('not', 'same', 'is not same');
    t.deepEqual({}, {}, 'is same object'); 
    t.notDeepEqual({}, {a: true}, 'is name same object'); 
    t.throws(Promise.reject(), 'is exception');
    t.notThrows(Promise.resolve(), 'is not exception');
    t.regex(a, /a/, 'is matching pattern');
    t.ifError(new Error(), 'is error');
});
```

### PR DESCRIPTION
**Pull request by jfmengels**
Adds `assertion-message` rule (fixes #59)

Haven't added any middleground option between `always` and `never`, as I can't see any that makes sense at the moment.

### CODE REVIEW

**Comment by jfmengels** [comment]
Ready for review :)

File: docs/rules/assertion-message.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+
+## Fail
+
+```js
+import test from 'ava';
+
+/*eslint ava/assertion-message: ["error", "always"]*/
+test(t => {
+	t.truthy(array.indexOf(value) !== -1);
```
Comment (sindresorhus) [inline docs/rules/assertion-message.md]: `t.true`

File: docs/rules/assertion-message.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+});
+
+/*eslint ava/assertion-message: ["error", "never"]*/
+test(t => {
+	t.truthy(array.indexOf(value) !== -1, 'value is not in array');
```
Comment (sindresorhus) [inline docs/rules/assertion-message.md]: `t.true`

File: docs/rules/assertion-message.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+Assertion messages are optional arguments that can be given to any assertion call to improve the error message, should the assertion fail. This rule either enforces or disallows the use of those messages.
+
+
+## Fail
+
+```js
+import test from 'ava';
+
+/*eslint ava/assertion-message: ["error", "always"]*/
```
Comment (sindresorhus) [inline docs/rules/assertion-message.md]:
```
/* eslint ava/assertion-message: ["error", "always"] */
```

File: docs/rules/assertion-message.md (hunk 1 above)
Comment (novemberborn) [inline docs/rules/assertion-message.md]: Wonder if we can lint for boolean expressions and require `t.true`?

Comment (sindresorhus) [inline docs/rules/assertion-message.md]: @novemberborn I was thinking about the other day too. Open a new issue and we can discuss there.

**Comment by sindresorhus** [comment]
LGTM when inline feedback is resolved.

File: docs/rules/assertion-message.md (hunk 1 above)
Comment (novemberborn) [inline docs/rules/assertion-message.md]: #77

**Comment by jfmengels** [comment]
Updated

**Comment by jfmengels** [comment]
It's not DiCaprio but that'll do I guess :p

**Comment by sindresorhus** [comment]
@jfmengels Dude, I've run out of DiCaprio celebration gif. @vdemedes Help us!!

**Pull request merged**